### ISSUE TITLE
Test rtl CLI with multi-stream workloads

### ISSUE THREAD

**Issue by sunway513** [sunway513/rocm-trace-lite#22]
## Goal

Validate that the `rtl trace` CLI works correctly when the traced workload uses multiple CUDA streams and threads. The existing multi-stream GPU tests (test_gpu_multithread.py) test the .so directly via HSA_TOOLS_LIB; this issue covers testing through the `rtl` CLI entry point.

### E1. Single process, multiple streams
```bash
rtl trace python3 -c "
import torch, threading
streams = [torch.cuda.Stream() for _ in range(4)]
def work(s):
    x = torch.randn(256,256,device='cuda')
    with torch.cuda.stream(s):
        for _ in range(50): x = x @ x
    s.synchronize()
threads = [threading.Thread(target=work, args=(s,)) for s in streams]
for t in threads: t.start()
for t in threads: t.join()
torch.cuda.synchronize()
"
```
**Assert**: trace.rpd has >= 200 ops, `rtl summary trace.rpd` shows correct kernel counts

### E2. Multi-GPU via rtl
```bash
rtl trace -o multi_gpu.rpd python3 -c "
import torch, threading
def work(gpu):
    x = torch.randn(256,256,device=f'cuda:{gpu}')
    for _ in range(50): x = x @ x
    torch.cuda.synchronize(gpu)
threads = [threading.Thread(target=work, args=(i,)) for i in range(2)]
for t in threads: t.start()
for t in threads: t.join()
"
```
**Assert**: `rtl info multi_gpu.rpd` shows ops on multiple gpuIds

### E3. Multi-stream + HIP Graph via rtl
```bash
rtl trace -o graph_stream.rpd python3 -c "
import torch
x = torch.randn(256,256,device='cuda')
s = torch.cuda.Stream()
with torch.cuda.stream(s): _ = x @ x
s.synchronize()
g = torch.cuda.CUDAGraph()
with torch.cuda.graph(g, stream=s): y = x @ x
for _ in range(50): g.replay()
s.synchronize()
# Also do eager work on another stream
s2 = torch.cuda.Stream()
with torch.cuda.stream(s2):
    for _ in range(20): z = x @ x
s2.synchronize()
"
```
**Assert**: no crash, trace has ops from both eager and graph paths

### E4. rtl summary and convert on multi-stream trace
- Run E1, then `rtl summary trace.rpd` — should show aggregated kernels
- Run `rtl convert trace.rpd -o trace.json` — should produce valid Perfetto JSON
- Verify JSON has events with correct pid/tid separation

## Implementation
Add `tests/test_cli_multistream.py` with GPU-required tests that use `subprocess.run` to invoke `python3 -m rocm_trace_lite.cli trace ...` and validate the output trace files.

## Acceptance criteria
- All 4 test cases pass on MI300X
- No crash, no data loss
- `rtl summary` and `rtl convert` produce correct output from multi-stream traces
